core/tx_pool.go

```diff
@@ -339,17 +341,6 @@ func (pool *TxPool) Pending() (map[common.Address]types.Transactions, error) {
 	pool.mu.Lock()
 	defer pool.mu.Unlock()
```

Comment:
Couldn't you rewrite `list.Flatten` like this - not even needing synchronization? 

```golang
// Flatten creates a nonce-sorted slice of transactions based on the loosely
// sorted internal representation. The result of the sorting is cached in case
// it's requested again before any modifications are made to the contents.
func (m *txSortedMap) Flatten() types.Transactions {
	// If the sorting was not cached yet, create and cache it
	if m.cache == nil {
		tempcache = make(types.Transactions, 0, len(m.items))
		for _, tx := range m.items {
			tempcache = append(tempcache, tx)
		}
		sort.Sort(types.TxByNonce(tempcache))
		m.cache = tempcache
	}

	tempcache := m.cache

	// Copy the cache to prevent accidental modifications
	txs := make(types.Transactions, len(tempcache))
	copy(txs, tempcache)
	return txs
}

```